runtime/lua/vim/lsp/util.lua

```diff
@@ -663,6 +662,26 @@ do
     end
   end
 
+  function M.buf_clear_references(bufnr)
+    validate { bufnr = {bufnr, 'n', true} }
+    api.nvim_buf_clear_namespace(bufnr, reference_ns, 0, -1)
+  end
+
+  function M.buf_highlight_references(bufnr, references)
+    validate { bufnr = {bufnr, 'n', true} }
+    for _, reference in ipairs(references) do
+      local start_pos = {reference["range"]["start"]["line"], reference["range"]["start"]["character"]}
+      local end_pos = {reference["range"]["end"]["line"], reference["range"]["end"]["character"]}
+      if reference["kind"] == protocol.DocumentHighlightKind.Text then
+        highlight_range(bufnr, reference_ns, "LspReferenceText", start_pos, end_pos)
+      elseif reference["kind"] == protocol.DocumentHighlightKind.Read then
+        highlight_range(bufnr, reference_ns, "LspReferenceRead", start_pos, end_pos)
+      elseif reference["kind"] == protocol.DocumentHighlightKind.Write then
+        highlight_range(bufnr, reference_ns, "LspReferenceWrite", start_pos, end_pos)
+      end
```

Comment:
call highlight_range outside of the if, just use the if to map the kind to the string LspReferenceXXXX

Reply:
yes, and then the if-else chain isn't needed at all. instead a temp map can be created and `protocol.DocumentHighlightKind` can be used as the lookup. Just like the other PR (which was merged before this one)